### PR TITLE
feat: use redis for sessions

### PR DESCRIPTION
**Pull request by lewxdev**

### CODE REVIEW

**Comment by lewxdev** [comment]
* **#29** <a href="https://app.graphite.dev/github/pr/lewxdev/mmmines/29?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#28** <a href="https://app.graphite.dev/github/pr/lewxdev/mmmines/28?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @lewxdev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by lewxdev** [comment]
something's wrong, the server doesn't load locally

File: app/utils/redis.ts [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+  const value = await redis.hget(sessionKey, id);
+  return value === "alive" || value === "dead" ? value : null;
+}
+
+export async function setSession(id: string, state: SessionState) {
+  const redis = getClient();
+  return redis.hset(sessionKey, id, state);
+}
+
+export async function resetSessions() {
```
Comment (drewradcliff) [inline app/utils/redis.ts:38]: we need to call this when the field is complete

**Review by drewradcliff — CHANGES_REQUESTED** [review]

File: app/utils/redis.ts (hunk 1 above)
Comment (lewxdev) [reply app/utils/redis.ts:38]: oop

**Review by drewradcliff — APPROVED** [review]
💪

**Pull request merged**